src/features/images/DeleteImagesByFilterModal.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  deleteImagesByFilterTask,
+  fetchTask,
+  selectDeleteImagesByFilterLoading,
+} from '../tasks/tasksSlice.js';
+import { selectActiveFilters } from '../filters/filtersSlice.js';
+import { SimpleSpinner, SpinnerOverlay } from '../../components/Spinner.jsx';
+import { ButtonRow, HelperText } from '../../components/Form.jsx';
+import Button from '../../components/Button.jsx';
+import NoneFoundAlert from '../../components/NoneFoundAlert.jsx';
+import { deleteImages, selectImagesCount, selectImagesCountLoading } from './imagesSlice.js';
+import { setModalOpen, setModalContent } from '../projects/projectsSlice.js';
+import { IMAGE_QUERY_LIMITS } from '../../config';
+import { selectWorkingImages } from '../review/reviewSlice.js';
+
+const DeleteImagesByFilterModal = () => {
+  const filters = useSelector(selectActiveFilters);
+  const deleteImagesByFilterLoading = useSelector(selectDeleteImagesByFilterLoading);
+  const imageCountIsLoading = useSelector(selectImagesCountLoading);
+  const workingImages = useSelector(selectWorkingImages);
+
+  const dispatch = useDispatch();
+
+  const imageCount = useSelector(selectImagesCount);
+
+  useEffect(() => {
+    if (deleteImagesByFilterLoading.isLoading && deleteImagesByFilterLoading.taskId) {
+      dispatch(fetchTask(deleteImagesByFilterLoading.taskId));
+    }
+  }, [deleteImagesByFilterLoading, dispatch]);
+
+  const handleDeleteImagesButtonClick = async () => {
+    const { isLoading, errors } = deleteImagesByFilterLoading;
+    const noErrors = !errors || errors.length === 0;
+    if (!isLoading && noErrors) {
+      if (imageCount <= IMAGE_QUERY_LIMITS[2]) {
+        dispatch(deleteImages(workingImages.map((i) => i._id)));
```

Comment:
Because we do paginated querying of the images, it's unlikely that `workingImages` will contain all of the images that match the current set of filters, so mapping over them to get their `_id`s probably won't give you an array that's fully representative of all the images that should be deleted. 

That said, TBH I think it's totally fine if we kick all delete-images-by-filters requests to the task and don't even bother with the `IMAGE_QUERY_LIMITS` threshold check and synchronous `deleteImages` option even if it's not that many images. I mostly just wanted to keep the sync `deleteImages` for situations when users were just deleting a handful quickly via multi-select.

Reply:
Implemented as you suggested